### PR TITLE
fix: :bug: Fix issues #228 #233 #235 #240

### PR DESCRIPTION
**Pull request by alograg**
Remove depreciated rules seen in issues, refer to discussion (https://github.com/green-code-initiative/ecoCode/discussions/271)

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -16,8 +16,8 @@ services:
       SONAR_ES_BOOTSTRAP_CHECKS_DISABLE: 'true'
     volumes:
       - type: bind
-        source: ./target/ecocode-java-plugin-1.5.1-SNAPSHOT.jar
-        target: /opt/sonarqube/extensions/plugins/ecocode-java-plugin-1.5.1-SNAPSHOT.jar
+        source: ./target/ecocode-java-plugin-1.5.2-SNAPSHOT.jar
```
Comment (dedece35) [inline docker-compose.yml:19]: this modifications are not done like this ... this kind of modification is a part of release process (described here https://github.com/green-code-initiative/ecoCode-common/blob/main/doc/INSTALL.md#howto-create-a-release-core-contributor-rights-needed )
modifications not needed

File: pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
     <groupId>io.ecocode</groupId>
     <artifactId>ecocode-java-plugin</artifactId>
-    <version>1.5.1-SNAPSHOT</version>
+    <version>1.5.2-SNAPSHOT</version>
```
Comment (dedece35) [inline pom.xml:7]: same remark like docker-compose modification

**Comment by dedece35** [comment]
Hi @alograg, 
thank you a lot for PR.
there are some lacks to remove deprecated rules : I want to write a DefinitionOfDone list (like https://github.com/green-code-initiative/ecoCode-common/blob/main/doc/starter-pack.md#definition-of-done-of-a-pr )  to check all work to do in order to delete deprecated rules.

here are lacks that I detected : 
- check exiting DoD list : https://github.com/green-code-initiative/ecoCode-common/blob/main/doc/starter-pack.md#definition-of-done-of-a-pr
   - create also a PR on repository https://github.com/green-code-initiative/ecoCode-java-test-project because it's a project to check all is ok on an real environment and not also with Unit Tests
   - `CHANGELOG.md` update
- add a new PR on `ecocode` repository to delete documentation rules (for example here : https://github.com/green-code-initiative/ecoCode/tree/main/ecocode-rules-specifications/src/main/rules/EC63 )
   - modification in this repository is very important because it is referenced on pom.xml of each plugin repository
   - a new release has to be done to be used in repository plugin (here is the process : https://github.com/green-code-initiative/ecoCode-common/blob/main/doc/INSTALL.md#howto-publish-a-new-version-of-ecocode-rules-specifications-on-maven-central )

**Review by dedece35 — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (alograg) [reply docker-compose.yml:19]: Ok, I'll revert the change.

File: pom.xml (hunk 2 above)
Comment (alograg) [reply pom.xml:7]: I will do the same revert

File: docker-compose.yml (hunk 1 above)
Comment (dedece35) [reply docker-compose.yml]: sorry, my bad, as we delete deprecated rules, we have to change minor version. you're rigth.
please update this version to 1.6.0-SNAPHOT please

**Review by dedece35 — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Deleted
 
-- Remove depreciated rules EC4, EC53, EC63 and EC75
+- Remove deprecated java rules
```
Comment (dedece35) [inline CHANGELOG.md:16]: please keep reference to rule IDs (simpler for future search in CHANGELOG :p )

**Review by dedece35 — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 4]
```diff
@@ -67,7 +67,7 @@
         <google.re2j>1.7</google.re2j>
 
         <!-- temporary version waiting for real automatic release in ecocode repository -->
-        <ecocode-rules-specifications.version>1.4.6</ecocode-rules-specifications.version>
+        <ecocode-rules-specifications.version>1.5.0-SNAPSHOT</ecocode-rules-specifications.version>
```
Comment (dedece35) [inline pom.xml:70]: this SNAPSHOT version is OK for development but we have to put here the released version 1.5.0 after making the new realese in "ecocode" repository and pushing it to maven central (with another process that we can find in documentation)

Comment (alograg) [reply pom.xml:70]: Ok, apply

**Comment by dedece35** [comment]
Hi @alograg,
please take into account all my previous unresolved notes.

Once, you will correct my only one review feedback on "ecoCode-rules-specifications" repository, I will create a new version 1.5.0 of this one.
Next, this PR will be ok on Checks Tab.

Thank you again !

File: docker-compose.yml (hunk 1 above)
Comment (alograg) [reply docker-compose.yml]: ok, i will change

File: pom.xml (hunk 2 above)
Comment (alograg) [reply pom.xml]: ok

File: CHANGELOG.md [hunk 5]
```diff
@@ -13,6 +13,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Deleted
 
+- Remove depreciated rules EC4, EC53, EC63 and EC75
```
[the remark this answers is not in the document]
Comment (alograg) [reply CHANGELOG.md]: ok

**Comment by alograg** [comment]
Hi @dedece35 
This is ok ?
Maybe I was wrong about something regarding the POM versions.

**Comment by dedece35** [comment]
Hi @alograg,
thank you again for your work.
the new 1.6.0 java release contains your work : check https://github.com/green-code-initiative/ecoCode-java/releases/tag/1.6.0

**Comment by alograg** [comment]
Thankyou @dedece35 

I am truly humbled and honored to have received such positive feedback from you.

It has been a pleasure to participate in the EcoCode community and I am grateful for the opportunity to contribute to such a valuable project.

I am committed to continuing to contribute to the community in any way that I can. I am always looking for new ways to improve my skills and knowledge, and I am eager to share what I have learned with others.

Thank you again for your support and encouragement. I am excited to continue working with you and the rest of the community.

**Pull request merged**